### ISSUE TITLE
Inclusion of any typed value disables checking of other values

### ISSUE THREAD

**Issue by chrisnojima** [microsoft/TypeScript#32165]
**TypeScript Version:**  
3.5.2

**Search Terms:**
array any untyped

**Code**

```ts
const poison: any = 3
const arr: Array<number> = [
    1,
    'two',
    poison
]
```

**Expected behavior:**
TS would complain about 'two' above. Commenting out `poison` does show the error. My goal was to type my array to catch bad values getting into it, yet a single `any`ed value disables me doing any kind of validation

**Actual behavior:**
TS ignores 'two'

**Playground Link:** 
https://www.typescriptlang.org/play/index.html#code/MYewdgzgLgBADiAlhcAuGBDMBPGBeGAZgChRJYMAnS9AQWo2wB4wBXAWwCMBTSgPnwwA2sRhiYARgA0o8QHIoAdxByZ4+EhRhiAXSA

**Related Issues:** 
Couldn't find it

**Comment by RyanCavanaugh** [comment]
Duplicate #26184
